leaving this in the comments so that I can go fix awx test once this merges `FAILED awx/main/tests/functional/dab_rbac/test_dab_rbac_api.py::test_assign_custom_delete_role`

--- a/ansible_base/lib/dynamic_config/dynamic_settings.py
+++ b/ansible_base/lib/dynamic_config/dynamic_settings.py
@@ -170,6 +170,8 @@ if 'ansible_base.rbac' in INSTALLED_APPS:
     # then the system must create one, this is used for naming the auto-created role
     ANSIBLE_BASE_ROLE_CREATOR_NAME = '{obj._meta.model_name}-creator-permission'
 
+    # Require change permission to get delete permission
+    ANSIBLE_BASE_DELETE_REQUIRE_CHANGE = True
     # Specific feature enablement bits
     # For assignments
     ANSIBLE_BASE_ALLOW_TEAM_PARENTS = True

--- a/ansible_base/rbac/validators.py
+++ b/ansible_base/rbac/validators.py
@@ -117,14 +117,18 @@ def validate_permissions_for_model(permissions, content_type: Optional[Model], m
                 raise ValidationError({'permissions': f'Permissions for model {role_model._meta.verbose_name} needs to include view, got: {display_perms}'})
 
     # check change edit permissions is given for every model that has update/delete/special actions listed
-    for cls, valid_model_permissions in permissions_by_model.items():
-        if 'delete' in cls._meta.default_permissions:
-            # import pdb; pdb.set_trace()
-            model_permissions = set(valid_model_permissions) & codename_list
-            non_add_model_permissions = {codename for codename in model_permissions if not is_add_perm(codename)}
-            if 'delete' in non_add_model_permissions and not any('change' in codename for codename in non_add_model_permissions):
-                display_perms = ', '.join(non_add_model_permissions)
-                raise ValidationError({'permissions': f'Permissions for model {role_model._meta.verbose_name} needs to include change, got: {display_perms}'})
+    if settings.ANSIBLE_BASE_DELETE_REQUIRE_CHANGE:
+        for cls, valid_model_permissions in permissions_by_model.items():
+            if 'delete' and 'change' in cls._meta.default_permissions:
+                model_permissions = set(valid_model_permissions) & codename_list
+                non_add_model_permissions = {codename for codename in model_permissions if not is_add_perm(codename)}
+                if any('delete' in codename for codename in non_add_model_permissions) and not any(
+                    'change' in codename for codename in non_add_model_permissions
+                ):
+                    display_perms = ', '.join(non_add_model_permissions)
+                    raise ValidationError(
+                        {'permissions': f'Permissions for model {role_model._meta.verbose_name} needs to include change, got: {display_perms}'}
+                    )
 
 
 def validate_codename_for_model(codename: str, model: Union[Model, Type[Model]]) -> str:

--- a/ansible_base/rbac/validators.py
+++ b/ansible_base/rbac/validators.py
@@ -116,6 +116,15 @@ def validate_permissions_for_model(permissions, content_type: Optional[Model], m
                 display_perms = ', '.join(non_add_model_permissions)
                 raise ValidationError({'permissions': f'Permissions for model {role_model._meta.verbose_name} needs to include view, got: {display_perms}'})
 
+    # check that edit permissions is given for every model that has update/delete/special actions listed
+    for cls, valid_model_permissions in permissions_by_model.items():
+        if 'change' in cls._meta.default_permissions:
+            model_permissions = set(valid_model_permissions) & codename_list
+            non_add_model_permissions = {codename for codename in model_permissions if not is_add_perm(codename)}
+            if non_add_model_permissions and not any('change' in codename for codename in non_add_model_permissions):
+                display_perms = ', '.join(non_add_model_permissions)
+                raise ValidationError({'permissions': f'Permissions for model {role_model._meta.verbose_name} needs to include edit, got: {display_perms}'})
+
 
 def validate_codename_for_model(codename: str, model: Union[Model, Type[Model]]) -> str:
     """Shortcut method and validation to allow action name, codename, or app_name.codename

--- a/ansible_base/rbac/validators.py
+++ b/ansible_base/rbac/validators.py
@@ -119,7 +119,7 @@ def validate_permissions_for_model(permissions, content_type: Optional[Model], m
     # check change edit permissions is given for every model that has update/delete/special actions listed
     if settings.ANSIBLE_BASE_DELETE_REQUIRE_CHANGE:
         for cls, valid_model_permissions in permissions_by_model.items():
-            if 'delete' and 'change' in cls._meta.default_permissions:
+            if 'delete' in cls._meta.default_permissions and 'change' in cls._meta.default_permissions:
                 model_permissions = set(valid_model_permissions) & codename_list
                 non_add_model_permissions = {codename for codename in model_permissions if not is_add_perm(codename)}
                 if any('delete' in codename for codename in non_add_model_permissions) and not any(

--- a/test_app/tests/rbac/test_validators.py
+++ b/test_app/tests/rbac/test_validators.py
@@ -106,3 +106,11 @@ class TestProhibitedRoleDefinitions:
         with pytest.raises(ValidationError) as exc:
             RoleDefinition.objects.create_from_permissions(name='system-inventory-viewer', permissions=['view_inventory'], content_type=None)
         assert 'System-wide roles are not enabled' in str(exc)
+
+@pytest.mark.django_db
+def test_no_delete_capability_without_change():
+    with pytest.raises(ValidationError) as exc:
+        RoleDefinition.objects.create_from_permissions(
+            name='anything', permissions=['view_credential', 'delete_credential'], content_type=permission_registry.content_type_model.objects.get_for_model(Credential)
+        )
+    assert 'Role definitions must have change permission' in str(exc)

